lib/awful/util.lua

```diff
@@ -40,8 +40,17 @@ local displayed_deprecations = {}
 -- @param[opt] see The message to a new method / function to use.
 -- @tparam table args Extra arguments
 -- @tparam boolean args.raw Print the message as-is without the automatic context
+-- @tparam integer args.deprecated_in Print the message only when Awesome's major version is equal to or greater than deprecated_in
 function util.deprecate(see, args)
     args = args or {}
+    -- If args.deprecated_in is defined, only print deprecation warning after the given major version
+    if args and args.deprecated_in then
+        local cur_ver, dep_ver = awesome.version, "v" .. tostring(args.deprecated_in) .. ".0"
```

Comment:
I would argue that it's better to pass in the whole version as-is (without the `v` then though?!)
(looking at the history of doing too much magic in here before)